purchase_advance_payment/wizard/purchase_advance_payment_wizard.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (C) 2021 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+
+from odoo import _, api, exceptions, fields, models
+from odoo.tools import float_compare
+
+
+class AccountVoucherWizardPurchase(models.TransientModel):
+
+    _name = "account.voucher.wizard.purchase"
+    _description = "Account Voucher Wizard Purchase"
+
+    order_id = fields.Many2one("purchase.order", required=True)
+    journal_id = fields.Many2one(
+        "account.journal",
+        "Journal",
+        required=True,
+        domain=[("type", "in", ("bank", "cash"))],
+    )
+    journal_currency_id = fields.Many2one(
+        "res.currency",
+        "Journal Currency",
+        store=True,
+        readonly=False,
+        compute="_compute_get_journal_currency",
+    )
+    currency_id = fields.Many2one("res.currency", "Currency", readonly=True)
+    amount_total = fields.Monetary(readonly=True)
+    amount_advance = fields.Monetary(
+        "Amount advanced", required=True, currency_field="journal_currency_id"
+    )
+    date = fields.Date(required=True, default=fields.Date.context_today)
+    currency_amount = fields.Monetary(
+        "Curr. amount", readonly=True, currency_field="currency_id"
+    )
+    payment_ref = fields.Char("Ref.")
+
+    @api.depends("journal_id")
+    def _compute_get_journal_currency(self):
+        for wzd in self:
+            wzd.journal_currency_id = (
+                wzd.journal_id.currency_id.id or self.env.user.company_id.currency_id.id
+            )
+
+    @api.constrains("amount_advance")
+    def check_amount(self):
+        if self.amount_advance <= 0:
+            raise exceptions.ValidationError(_("Amount of advance must be positive."))
+        if self.env.context.get("active_id", False):
+            self.onchange_date()
+            if (
+                float_compare(
+                    self.currency_amount,
+                    self.order_id.amount_residual,
+                    precision_digits=2,
+                )
+                > 0
+            ):
```

Comment:
```suggestion
            if self.currency_id.compare_amounts(self.currency_amount, self.order_id.amount_residual) > 0:
```

This respects the currency rounding, instead of hardcoding to 2 digits

Reply:
Updated, Thanks